frontends/p4/sideEffects.cpp

```diff
@@ -172,6 +173,15 @@ class DismantleExpression : public Transform {
             // the MethodCallExpression.
             if (TableApplySolver::isHit(expression, refMap, typeMap) ||
                 TableApplySolver::isMiss(expression, refMap, typeMap)) {
+                if (!getContext()) {
```

Comment:
is this the problem or is the bug where the second table reference is introduced?
In general, the number of table invocations on each program path should not be changed by the compiler, so perhaps there is a bug somewhere?
Which pass introduces an extra table access?

Reply:
ActionSynthesis creates new tables and actions for assignments in a control block.  So we end up with

    if (table.apply().hit) {
        t1.apply();  // run action { tmp = true; }
    } else {
        t2.apply(); // run action { tmp = false; }
    }
    if (tmp) {
        ...

and the apply().hit is back inside an if, and we have two extra (useless) tables.